asciidoctorj-cli/src/main/java/org/asciidoctor/cli/jruby/AsciidoctorInvoker.java

```diff
@@ -150,46 +137,17 @@ private URLClassLoader createUrlClassLoader(List<String> classPaths) {
         return new URLClassLoader(cpUrls.toArray(new URL[cpUrls.size()]));
     }
 
-    private void convertInput(Asciidoctor asciidoctor, Options options, List<File> inputFiles) {
+    private void convertInput(JRubyAsciidoctor asciidoctor, AsciidoctorCliOptions cliOptions, List<File> inputFiles) throws IOException {
+        Ruby ruby = JRubyRuntimeContext.get(asciidoctor);
+        RubyHash opts = cliOptions.parse(ruby);
 
-        if (inputFiles.size() == 1 && "-".equals(inputFiles.get(0).getName())) {
-            asciidoctor.convert(readInputFromStdIn(), options);
-            return;
-        }
+        new RubyGemsPreloader(asciidoctor.getRubyRuntime()).preloadRequiredLibrariesCommandLine(opts);
 
-        options.setMkDirs(true);
-
-        findInvalidInputFile(inputFiles)
-                .ifPresent(inputFile -> {
-                    System.err.println("asciidoctor: FAILED: input file(s) '"
-                            + inputFile.getAbsolutePath()
-                            + "' missing or cannot be read");
-                    throw new IllegalArgumentException(
-                            "asciidoctor: FAILED: input file(s) '"
-                                    + inputFile.getAbsolutePath()
-                                    + "' missing or cannot be read");
-                });
-
-        final Optional<File> toDir = getAbsolutePathFromOption(options, Options.TO_DIR);
-        final Optional<File> srcDir = getAbsolutePathFromOption(options, Options.SOURCE_DIR);
-
-        inputFiles.forEach(inputFile -> {
-            if (toDir.isPresent() && srcDir.isPresent()) {
-                if (inputFile.getAbsolutePath().startsWith(srcDir.get().getAbsolutePath())) {
-                    String relativePath = srcDir.get().toURI().relativize(inputFile.getParentFile().getAbsoluteFile().toURI()).getPath();
-                    String absolutePath = new File(toDir.get(), relativePath).getAbsolutePath();
-                    options.setToDir(absolutePath);
-                }
-            }
-            asciidoctor.convertFile(inputFile, options);
-        });
-    }
+        opts.put(ruby.newSymbol("input_files"), inputFiles.stream().map(f -> ruby.newString(f.getPath())).collect(Collectors.toList()));
```

Comment:
Question: I understand `input_files` is an Asciidoctor Ruby feature and we don't need any of the validations we did? If so, it's fine, but we should have some tests to validate the behauviour imo.

Reply:
I would leave the interpretation of these files up to Asciidoctor Ruby.
We should not make any assumptions whether these files exist or not.
For example, the file `-` used for reading the input from stdin usually does not exist.